### ISSUE TITLE
non-conformant zip files in 1.12.1 release?

### ISSUE THREAD

**Issue by 3b** [Clozure/ccl#400]
The 1.12.1 windows .zip files (https://github.com/Clozure/ccl/releases/download/v1.12.1/ccl-1.12.1-windowsx86.zip and https://github.com/Clozure/ccl/releases/download/v1.12.1/windowsx86.zip) seem to be using `\` as path separator instead of `/` as required by zip spec.

This confuses the roswell (roswell/roswell#491), preventing it from installing current CCL. Might also be causing the problems with 7-zip mentioned in #385.

Microsoft tools extract it without complaint, and `unzip` with only a warning, so contents are probably otherwise OK and zip just needs to be rebuilt.

**Comment by xrme** [comment]
What unzip program should I use to test that I made the zip file correctly?

**Comment by 3b** [comment]
should be able to tell if roswell will be able to use it with

```lisp
(ql:quickload 'zip)
(zip:unzip "/tmp/ccl-1.12.1-windowsx86.zip"
           "/tmp/ccltmp/"
            :IF-EXISTS :ERROR :VERBOSE t :FORCE-UTF-8 NIL)
```

there is a command line `zipdetails` that will print very verbose info including the names stored in the file, i have it on my msys2 and ubuntu systems, but not sure exactly which packages provide it.

the `unzip` program shows it as well, either in names with `-t` or as warning when decompressing
```
$ unzip -t ccl-1.12.1-windowsx86.zip
Archive:  ccl-1.12.1-windowsx86.zip
    testing: ccl\win32-headers\       OK
    testing: ccl\win64-headers\       OK
    testing: ccl\.gitattributes       OK
    testing: ccl\.gitignore           OK
...

Archive:  ccl-1.12.1-windowsx86.zip
warning:  ccl-1.12.1-windowsx86.zip appears to use backslashes as path separators
   creating: ccl/win32-headers/
   creating: ccl/win64-headers/
  inflating: ccl/.gitattributes
  inflating: ccl/.gitignore
...

```

**Comment by asmaloney** [comment]
@xrme Any way the current `ccl-1.12.1-windowsx86.zip` could be fixed and re-uploaded? (Or maybe do a new release with latest changes & fix it then?)

When you unzip it all the files are in one dir because of this path issue:

<img width="456" alt="Screen Shot 2022-06-10 at 22 51 08 PM" src="https://user-images.githubusercontent.com/391371/173169925-6cf1d330-9a99-4a57-89ca-c881a2537c39.png">

This is preventing my project from using it ☹️

**Comment by asmaloney** [comment]
@xrme Would it be possible to get this repacked or a new release to fix it please? Standard tooling for zips (e.g. go's std lib which is what I'm using) won't read this file properly.

I tried to back off to 1.12, but ran into [this problem](https://github.com/Clozure/ccl/issues/311), so I'm stuck...

Thanks!

**Comment by edoneel** [comment]
Hi

I have rebuilt them and uploaded them to the URLs below.  I tested them with (zip:unzip ...)

http://www.pckswarms.ch/ccl/ccltmp/ccl-1.12.1-windowsx86-beo.zip

http://www.pckswarms.ch/ccl/ccltmp/windowsx86-beo.zip

You have to decide if this is a good idea to download them or not but I can not upload them to the real place.

cheers

bruce

**Comment by asmaloney** [comment]
Thanks Bruce! That will be helpful for testing.

My [research tool](https://github.com/asmaloney/gactar) automatically downloads the releases from the ccl repo, so it would be helpful to get a fixed version put up here. I recently switched from SBCL because this looked easier to download & use in my context.

**Comment by ntrocado** [comment]
> You have to decide if this is a good idea to download them or not but I can not upload them to the real place.

I trusted and it works.

**Comment by asmaloney** [comment]
Looks like this is fixed with [1.12.2](https://github.com/Clozure/ccl/releases/tag/v1.12.2).

**Comment by xrme** [comment]
When I used built-in Windows functionality to make the zip files, it used backslashes as the path separator. I made the 1.12.2 zip files with the info-zip `zip` program this time, and it used the forward slash.

Now that the 1.12.2 release has correct zip files, I'll close this issue.
